editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx

```diff
@@ -214,16 +189,137 @@ describe("validation and error handling", () => {
 test.todo("an error displays if the maximum number of items is exceeded");
 
 describe("basic interactions - happy path", () => {
-  test.todo("adding an item to the map adds a feature tab");
-  // add feature, see a tab (one feature only)
-  test.todo("a user can input details on a single feature and submit");
-  // only one feature, fill out form, submit
-  test.todo("adding multiple features to the map adds multiple feature tabs");
-  // add more than one feature, see multiple tabs
-  test.todo("a user can input details on multiple features and submit");
+  it("adding an item to the map adds a feature tab", async () => {
+    const { getByTestId } = setup(<MapAndLabel {...props} />);
+    const map = getByTestId("map-and-label-map");
+
+    addFeaturesToMap(map, [point1]);
+
+    const firstTabPanel = getByTestId("vertical-tabpanel-0");
+
+    expect(firstTabPanel).toBeVisible();
+  });
+
+  it("a user can input details on a single feature and submit", async () => {
+    const { getByTestId, user } = setup(<MapAndLabel {...props} />);
+
+    const map = getByTestId("map-and-label-map");
+
+    addFeaturesToMap(map, [point1]);
+
+    const firstTabPanel = getByTestId("vertical-tabpanel-0");
+
+    expect(firstTabPanel).toBeVisible();
+
+    await fillOutForm(user);
+
+    await clickContinue(user);
+
+    await checkErrorMessagesEmpty();
+  });
+
+  it("adding multiple features to the map adds multiple feature tabs", async () => {
+    const { getByTestId, queryByRole } = setup(<MapAndLabel {...props} />);
+    const map = getByTestId("map-and-label-map");
+
+    addMultipleFeatures([point1, point2, point3]);
+
+    // vertical side tab query
+    const firstTab = queryByRole("tab", { name: /Tree 1/ });
+    const secondTab = queryByRole("tab", { name: /Tree 2/ });
+    const thirdTab = queryByRole("tab", { name: /Tree 3/ });
+    const fourthTab = queryByRole("tab", { name: /Tree 4/ });
+
+    // check the right amount are in the document
+    expect(firstTab).toBeInTheDocument();
+    expect(secondTab).toBeInTheDocument();
+    expect(thirdTab).toBeInTheDocument();
+    expect(fourthTab).not.toBeInTheDocument();
+
+    expect(thirdTab).toHaveAttribute("aria-selected", "true");
+    expect(secondTab).toHaveAttribute("aria-selected", "false");
+    expect(firstTab).toHaveAttribute("aria-selected", "false");
+  });
+
+  it("a user can input details on multiple features and submit", async () => {
+    const { getByTestId, getByRole, user, getAllByTestId } = setup(
+      <MapAndLabel {...props} />,
+    );
+    const map = getByTestId("map-and-label-map");
```

Comment:
```suggestion
  it("a user can input details on multiple features and submit", async () => {
    const { getByTestId, getByRole, user } = setup(
      <MapAndLabel {...props} />,
    );
```

nit: A few unused variables here and there in this test file, have a quick scan through 👍